### ISSUE TITLE
[ARITH] Enhance CanonicalSimplify(x*tvm.max(x, 1) - x*tvm.max(x, 1)) 

### ISSUE THREAD

**Issue by derisavi** [dmlc/tvm#2248]
During debugging some experimental code, I found out about a limitation in `CanonicalSimplify`. I reduced the testcase to the following:
```python
import tvm
x = tvm.var("x")
ret = tvm.ir_pass.CanonicalSimplify(x*tvm.max(x, 1) - x*tvm.max(x, 1))
print(ret)
```
I get the same expression as a result: `((x*max(x, 1)) - (x*max(x, 1)))`. 

@tqchen (or other folks) is this something that can be fixed with fairly low effort? Any tips on how this can be fixed is appreciated.

**Comment by tqchen** [comment]
The only chance to make this happen is to map ```x*max(x, 1)``` and ```x*max(x, 1)``` to the same expression. We did this to some extent, so to debug how to make the improvement, please check if ```max(x, 1)``` does the binary caching wrt to its input and possibly add that

**Comment by tqchen** [comment]
solved by #2261
